lib/LDA2Vec.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python
+import numpy
+from lib.content_based import ContentBased
+from sklearn.feature_extraction.text import CountVectorizer
+
+import chainer
+from chainer import optimizers
+from lda2vec import preprocess
+from lda2vec import utils
+from lda2vec_model import LDA2Vec
+
+
+class LDA2VecRecommender(ContentBased):
+    def __init__(self, abstracts, evaluator, config, verbose=False):
+        super(LDA2VecRecommender, self).__init__(abstracts, evaluator, config, verbose)
+
+    def train(self, n_iter=5):
+        """
+        Train the LDA2Vec model.
+        """
+        skip_character = -2
+        n_units = max(map(lambda doc: len(doc.split(' ')), self.abstracts)) + 1
+        arr, vocab = preprocess.tokenize(map(unicode, self.abstracts), n_units, skip=skip_character)
+        doc_ids = []
+        flattened = []
+        for doc_id, words in enumerate(arr):
+            for word_id in words:
+                if word_id != skip_character:
+                    doc_ids.append(doc_id)
+                    flattened.append(word_id)
+        flattened = numpy.array(flattened, dtype='int32')
+        doc_ids = numpy.array(doc_ids, dtype='int32')
+        n_vocab = flattened.max() + 1
+        tok_idx, freq = numpy.unique(flattened, return_counts=True)
+        term_frequency = numpy.zeros(n_vocab, dtype='int32')
+        term_frequency[tok_idx] = freq
```

Comment:
The main reason is the format that LDA2Vec models needs, was necessary to have the two lists flattened, doc_ids. And also the format of the term_frequency. The CountVectorizer returns a (sparse) matrix of counts which is a different format.